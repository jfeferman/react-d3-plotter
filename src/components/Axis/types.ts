export type AxisProps = {
  orient: string,
  scale: Function,
  translate: string,
}
