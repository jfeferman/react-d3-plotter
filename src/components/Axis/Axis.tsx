import { useEffect } from 'react'
import * as d3 from 'd3'

export type AxisProps = {
  orient: string
  scale: () => void
  translate: string
}

export const Axis = ({ orient, scale, translate }: AxisProps) => {
  const axisContainer = React.createRef<SVGGElement>()

  useEffect(() => {
    renderAxis()
  }, [])

  const renderAxis = () => {
    const node = axisContainer.current
    let axis
    if (orient === 'bottom') {
      axis = d3.axisBottom().ticks(5).scale(scale)
    } else {
      axis = d3.axisLeft().ticks(5).scale(scale)
    }

    d3.select(node).call(axis)
  }

  return <g className="axis" ref={axisContainer} transform={translate} />
}
