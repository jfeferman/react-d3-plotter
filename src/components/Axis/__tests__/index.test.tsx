import * as React from 'react'
import { shallow } from 'enzyme'
import Axis from '../Axis'
import {
  getXScale,
  getYScale,
} from '../../../utils/testUtil'

describe('components/Axis', () => {
  it('checks the x axis', () => {
    const wrapper = shallow(
      <Axis
        orient="bottom"
        scale={getXScale()}
        translate="0"
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('checks the y axis', () => {
    const wrapper = shallow(
      <Axis
        orient=""
        scale={getYScale()}
        translate="0"
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
