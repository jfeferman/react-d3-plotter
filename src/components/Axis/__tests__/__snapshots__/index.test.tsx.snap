// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Axis checks the x axis 1`] = `
<g
  className="axis"
  transform="0"
/>
`;

exports[`components/Axis checks the y axis 1`] = `
<g
  className="axis"
  transform="0"
/>
`;
